pkg/communication/sdk_client_protocol.go

```diff
@@ -0,0 +1,157 @@
+
+	endMsg := &EndpointMessage{
+		ProtobufMessage: request,
+	}
+	endpoint.Send(endMsg)
+	defer close(endpoint.MessageReceiver())
+	// Wait for the response to be received by the transport and then parsed and put on the endpoint's message channel
+	serverMsg, ok := <-endpoint.MessageReceiver()
+	if !ok {
+		fmt.Println("[SdkClientProtocol][" + endpoint.GetName() + "] : Endpoint Receiver channel is closed")
+		endpoint.CloseEndpoint()
+		return false
+	}
+	fmt.Printf("[SdkClientProtocol] ["+endpoint.GetName()+"] : Received: %s\n", serverMsg)
+
+	// Handler response
+	negotiationResponse := protoMsg.NegotiationMsg
+	negotiationResponse.GetNegotiationResult()
+
+	if negotiationResponse.GetNegotiationResult().String() != version.NegotiationAnswer_ACCEPTED.String() {
+		fmt.Println("[SdkClientProtocol] Protocol version negotiation failed",
+			negotiationResponse.GetNegotiationResult().String()+") :"+negotiationResponse.GetDescription())
+		return false
+	}
+	fmt.Println("[SdkClientProtocol] Protocol version is accepted by server: {}", negotiationResponse.GetDescription())
+	endpoint.CloseEndpoint()
+	return true
+}
+
+func getSpecificationVersion() string {
+	specificationVersion := "5.8.0-SNAPSHOT"
+	//TODO: final String specificationVersion = CommonDTO.class.getPackage().getSpecificationVersion();
+	if specificationVersion == "" {
+		glog.V(3).Infof("Specification-Version is null for CommonDTO class. ",
+			"This should not be so in production run")
+		return ""
+	}
+	return specificationVersion
+}
+
+// ======================= Registration ============================
+// Send registration message
+func (clientProtocol *SdkClientProtocol) HandleRegistration(transport ITransport) bool {
+	containerInfo, err := clientProtocol.MakeContainerInfo()
+	if err != nil {
+		glog.Error("Error creating ContainerInfo")
+		return false
+	}
+
+	glog.V(3).Infof("[SdkClientProtocol] Send registration message: %+v", containerInfo)
+
+	// Create Protobuf Endpoint to send and handle registration messages
+	protoMsg := &RegistrationResponse{}
+	endpoint := CreateClientProtobufEndpoint2("RegistrationEndpoint", transport, protoMsg)
+
+	endMsg := &EndpointMessage{
+		ProtobufMessage: containerInfo,
+	}
+	endpoint.Send(endMsg)
+	fmt.Println("[SdkClientProtocol][" + endpoint.GetName() + "] : Waiting for registration response")
+	// Wait for the response to be received by the transport and then parsed and put on the endpoint's message channel
+	serverMsg, ok := <-endpoint.MessageReceiver()
+	if !ok {
+		fmt.Println("[" + endpoint.GetName() + "] : Endpoint Receiver channel is closed")
+		endpoint.CloseEndpoint()
+		return false
+	}
+	fmt.Printf("[SdkClientProtocol] ["+endpoint.GetName()+"] : Received: %s\n", serverMsg)
+
+	// Handler response
+	registrationResponse := protoMsg.RegistrationMsg
+	if registrationResponse == nil {
+		fmt.Println("[SdkClientProtocol] Probe registration failed, null ack")
```

Comment:
Can we have a more detailed error handling here to handle the registration response, and have a case by case handler for it. Basically this is the very first step to communicate with Turbo, and if this failed, what should the customer do here?